core/middlewares/with-routes.ts

```diff
@@ -249,21 +249,26 @@ export const withRoutes: MiddlewareFactory = () => {
     // Use 301 status code as it is more universally supported by crawlers
     const redirectConfig = { status: 301 };
 
-    if (route?.redirect) {
+    if (route?.node && route.node.__typename === 'Product') {
+      // If there's a valid Product node, process it as a normal product page
+      const productUrl = new URL(`/${locale}/product/${route.node.entityId}`, request.url);
+  
+      return NextResponse.rewrite(productUrl);
+    } else if (route?.redirect) {
       switch (route.redirect.to.__typename) {
         case 'ManualRedirect': {
           // For manual redirects, redirect to the full URL to handle cases
           // where the destination URL might be external to the site
           return NextResponse.redirect(route.redirect.toUrl, redirectConfig);
         }
-
+  
         default: {
```

Comment:
Instead of adding additional logic before this switch statement, what if we just `break` on the `default` case and fallback to the logic below? It seems like if it's an internal redirect `node` will be populated which would fall into the normal `rewrite` logic 🤔 

This is what I mean https://github.com/bigcommerce/catalyst/compare/fix/redirects

Curious if that'll fix this issue?

Reply:
Can confirm this fixes the infinite redirect, but I don't know what side effect it has or how to test that accurately

Reply:
@willPrattUPL can you list out what edge cases now return 404's when using the `break` method?

Reply:
with your early break implementation

## gets past infinite redirect & loads normally:



**/uplift-v2-standing-desk-v2-or-v2-commercial/**

Value: {
  "route": {
    "redirect": {
      "__typename": "Redirect",
      "to": {
        "__typename": "ProductRedirect"
      },
      "toUrl": "https://www.upliftdesk.com/uplift-v2-standing-desk-v2-or-v2-commercial/"
    },
    "node": {
      "__typename": "Product",
      "id": "UHJvZHVjdDo4OTA=",
      "entityId": 890
    }
  },
  "expiryTime": 1728479934882
}



**/uplift-v2-l-shaped-standing-desk/**

Value: {
  "route": {
    "redirect": {
      "__typename": "Redirect",
      "to": {
        "__typename": "ProductRedirect"
      },
      "toUrl": "https://www.upliftdesk.com/uplift-v2-l-shaped-standing-desk/"
    },
    "node": {
      "__typename": "Product",
      "id": "UHJvZHVjdDo4NTk=",
      "entityId": 859
    }
  },
  "expiryTime": 1728480227568
}




**/uplift-v2-standing-desk-frame/**

Value: {
  "route": {
    "redirect": {
      "__typename": "Redirect",
      "to": {
        "__typename": "ProductRedirect"
      },
      "toUrl": "https://www.upliftdesk.com/uplift-v2-standing-desk-frame/"
    },
    "node": {
      "__typename": "Product",
      "id": "UHJvZHVjdDoyNTQ=",
      "entityId": 254
    }
  },
  "expiryTime": 1728480292572
}


--------------------------------------------------------------------------------

## 404:

**/height-adjustable-standing-desk-with-l-shaped-top/**

[BigCommerce] KV MGET - Keys: [redact]/height-adjustable-standing-desk-with-l-shaped-top/,[redact] - Value: [
  {
    "route": {
      "redirect": {
        "__typename": "Redirect",
        "to": {
          "__typename": "ProductRedirect"
        },
        "toUrl": "https://www.upliftdesk.com/uplift-v2-l-shaped-standing-desk/"
      },
      "node": null
    },
    "expiryTime": 1728479977089
  },
  {
    "status": "LAUNCHED",
    "expiryTime": 1728478477014
  }
]
Debugger attached.
[BigCommerce] query RootLayoutMetadataQuery - 8ms - complexity 1006
[BigCommerce] query NotFoundQuery - 2ms - complexity 5400
[BigCommerce] query LayoutQuery - 186ms - complexity 2225
[BigCommerce] query RootLayoutMetadataQuery - 2ms - complexity 1006
 GET /height-adjustable-standing-desk-with-l-shaped-top/ 404 in 1857ms
Waiting for the debugger to disconnect...

Reply:
Hmmm `node` coming back `null` for a `ProductRedirect` seems like a bug on our API 🤔 I'm going to check with the team if that's the case.

Reply:
Hey @willPrattUPL just chatted with the team and saw we have an option on the `route` node that might fix that last 404 edge case. Try setting the `redirectBehavior` to `FOLLOW` in the query along with the `break;` statement flow to see if that fixes the issue.
![Screenshot 2024-10-11 at 13 55 12](https://github.com/user-attachments/assets/c7ef957e-e6e6-49ee-9a14-93395ba091d8)

Reply:
So just to clarify, you are saying to add this to the query in get-route:

      route(path: $path, redirectBehavior: FOLLOW) {
      
      ?
      
      If so, that does indeed fix the edge case, as well

Reply:
Yea! That's great then! I think we update this PR to that instead, that way we don't need to duplicate that similar logic.